cmake/modules/SearchRootCoreDeps.cmake

```diff
@@ -14,77 +14,100 @@ if(pyroot_experimental)
 endif()
 
 if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.14)
-  # - Check if the PYTHON_EXECUTABLE deprecated variable was passed by the
-  # user; if so, check weather it points to Python 2 or 3 and set the
-  # appropriate Python{X}_EXECUTABLE variable
-  # - Look for Python3 and set the deprecated variables to the ones set
-  # automatically by find_package(Python3 ...)
-  # - Look for Python2 and set the deprecated variables to the ones set
-  # automatically by find_package(Python2 ...) ONLY IF PYTHON3 WASN'T FOUND
-  if(PYTHON_EXECUTABLE)
-    execute_process(COMMAND ${PYTHON_EXECUTABLE} -c "import sys;print(sys.version_info[0])"
+
+  if(PYTHON_EXECUTABLE OR Python_EXECUTABLE)
+    # - Check if the PYTHON_EXECUTABLE deprecated variable or Python_EXECUTABLE
+    # was passed by the user; if so, PyROOT will be built with a single version
+    if(PYTHON_EXECUTABLE)
+      set(Python_EXECUTABLE "${PYTHON_EXECUTABLE}")
+    endif()
+    execute_process(COMMAND ${Python_EXECUTABLE} -c "import sys;print(sys.version_info[0])"
                     OUTPUT_VARIABLE PYTHON_PREFER_VERSION
                     ERROR_VARIABLE PYTHON_PREFER_VERSION_ERR)
     if(PYTHON_PREFER_VERSION_ERR)
-      message(WARNING "Unable to determine version of ${PYTHON_EXECUTABLE}: ${PYTHON_PREFER_VERSION_ERR}")
+      message(WARNING "Unable to determine version of ${Python_EXECUTABLE}: ${PYTHON_PREFER_VERSION_ERR}")
     endif()
     string(STRIP "${PYTHON_PREFER_VERSION}" PYTHON_PREFER_VERSION)
-    set(Python${PYTHON_PREFER_VERSION}_EXECUTABLE "${PYTHON_EXECUTABLE}")
-  endif()
+    find_package(Python COMPONENTS Interpreter Development NumPy)
+    if(Python_Development_FOUND)
+      set(PYTHON_EXECUTABLE "${Python_EXECUTABLE}" CACHE INTERNAL "" FORCE)
+      set(PYTHON_INCLUDE_DIRS "${Python_INCLUDE_DIRS}" CACHE INTERNAL "" FORCE)
+      set(PYTHON_LIBRARIES "${Python_LIBRARIES}" CACHE INTERNAL "" FORCE)
+      set(PYTHON_VERSION_STRING "${Python_VERSION}" CACHE INTERNAL "" FORCE)
+      set(PYTHON_VERSION_MAJOR "${Python_VERSION_MAJOR}" CACHE INTERNAL "" FORCE)
+      set(PYTHON_VERSION_MINOR "${Python_VERSION_MINOR}" CACHE INTERNAL "" FORCE)
+      set(PYTHON_UNDER_VERSION_STRING "${Python_VERSION_MAJOR}_${Python_VERSION_MINOR}")
+      set(NUMPY_FOUND ${Python_NumPy_FOUND} CACHE INTERNAL "" FORCE)
+      set(NUMPY_INCLUDE_DIRS "${Python_NumPy_INCLUDE_DIRS}" CACHE INTERNAL "" FORCE)
+    endif()
+    if(DEFINED Python_VERSION AND "${Python_VERSION}" VERSION_LESS "2.7")
+      message(WARNING "Ignoring Python installation: unsupported version ${Python_VERSION} (version>=2.7 required)")
```

Comment:
IIRC, here we just give a warning message, but then somewhere else we will fail if we have only found an unsupported version?

Reply:
yes it should be related to scientifix linux (?). should I throw something more than a warning?

Reply:
Scientific Linux? I don't understand :confused: 

My question is: here we just give a warning, but somewhere else we make the configuration fail, right?

Reply:
wasn't this a warning that we added after a ticket was opened concerning build failing with Python < 2.7 in some SL6 build?

I have to check if we fail somewhere else because I don't remember

Reply:
No actually I don't think we do, because here 
https://github.com/root-project/root/commit/f62fb2ac5cce0f7576111ae0a0ecf685f4e19b0c#diff-5a4af40c5dab2eba7ad500714197987a
it was changed from fatal_error to warning but for different reasons.
So yes, probably we should put a fatal error here

Reply:
Well here we have two options:
1. We fail because the user specified a Python version that we can't build with, so we need to stop and tell her.
2. We give the warning and we go on building ROOT without PyROOT.

I think option 1 is the best here, opinions?

Reply:
I'm fine with the first. So fatal error?

Reply:
@oshadura what do you think, we go for 1?

Reply:
@oshadura here

Reply:
Yup! Lets do it (make a separate commit so we can revert it)